astropy/wcs/utils.py

```diff
@@ -825,3 +826,237 @@ def pixel_to_pixel(wcs_in, wcs_out, *inputs):
+        coordinates will be used as the projection point. To specify an exact
+        point for the projection, a Skycoord object with a coordinate pair can
+        be passed in. For consistency, the units and frame of these coordinates
+        will be transformed to match 'world_coords' if they don't.
+    projection : str or `~astropy.wcs.WCS`
+        Three letter projection code, of any of standard projections defined 
+        in the FITS WCS standard. Optionally, a WCS object with projection 
+        keywords set may be passed in. 
+    sip_degree : int
+        If set to a non-zero integer value, will fit SIP of degree `sip_degree` 
+        to model geometric distortion. 
+
+    Returns
+    -------
+    wcs : `~astropy.wcs.WCS`
+        The best-fit WCS to the points given.
+    """
+
+    from astropy.coordinates import SkyCoord # here to avoid circular import
+    import astropy.units as u
+    from scipy.optimize import least_squares
+    from .wcs import Sip
+    import copy
+
+    xp, yp = xy
+    lon, lat = world_coords.data.lon.deg, world_coords.data.lat.deg
+
+    # verify input
+    if (proj_point != 'center') and (type(proj_point) != type(world_coords)):
+        raise ValueError("proj_point must be set to 'center', or an" +
+                         "`~astropy.coordinates.SkyCoord` object with " +
+                         "a pair of points.")
+    if proj_point != 'center':
+        assert proj_point.size == 1
+
+    proj_codes = [
+        'AZP', 'SZP', 'TAN', 'STG', 'SIN', 'ARC', 'ZEA', 'AIR', 'CYP',
+        'CEA', 'CAR', 'MER', 'SFL', 'PAR', 'MOL', 'AIT', 'COP', 'COE',
+        'COD', 'COO', 'BON', 'PCO', 'TSC', 'CSC', 'QSC', 'HPX', 'XPH'
+    ]
+    if type(projection) == str:
+        if projection not in proj_codes:
+            raise ValueError("Must specify valid projection code from list of " 
+                             + "supported types: ", ', '.join(proj_codes))
+        # empty wcs to fill in with fit values
+        wcs = celestial_frame_to_wcs(frame=world_coords.frame,
+                                     projection=projection)
+    else: #if projection is not string, should be wcs object. use as template.
+        wcs = copy.deepcopy(projection)
+        wcs.cdelt = (1., 1.) # make sure cdelt is 1
+        wcs.sip = None
+
+    # Change PC to CD, since cdelt will be set to 1
+    if wcs.wcs.has_pc():
+        wcs.wcs.cd = wcs.wcs.pc
+        wcs.wcs.__delattr__('pc')
+
+    if (type(sip_degree) != type(None)) and (type(sip_degree) != int):
+       raise ValueError("sip_degree must be None, or integer.")
+
+    # set pixel_shape to span of input points
+    wcs.pixel_shape = (max(xp)-min(xp), max(yp)-min(yp))
+
+    # determine CRVAL from input
+    close = lambda l, p: p[np.where(np.abs(l) == min(np.abs(l)))[0][0]]
+    if str(proj_point) == 'center':  # use center of input points
+        sc1 = SkyCoord(min(lon)*u.deg, max(lat)*u.deg)
```

Comment:
As before, use ``.min()`` and ``.max()`` if these are arrays (comment applies to other instances)